source/state_representation/test/tests/test_parameter_map.cpp

```diff
@@ -0,0 +1,44 @@
+#include <gtest/gtest.h>
+
+#include "state_representation/parameters/ParameterMap.hpp"
+#include "state_representation/exceptions/InvalidParameterException.hpp"
+
+using namespace state_representation;
+
+class TestParameterMap : public ParameterMap {
+public:
+  explicit TestParameterMap(const ParameterInterfaceList& parameters) : ParameterMap(parameters) {}
+
```

Comment:
Related to the comment about clearly testing the assertion behavior, if you elevate the visibility of the assertion method you can test it directly. That way you can have two separate test cases:

- Does `assert_parameter_valid` throw when the name or type is invalid?
- Is `validate_and_set_parameter` called when I set one or more parameters?

```suggestion
class TestParameterMap : public ParameterMap {
public:
  explicit TestParameterMap(const ParameterInterfaceList& parameters) : ParameterMap(parameters) {}
  
  using ParameterMap::assert_parameter_valid(const std::shared_ptr<ParameterInterface>& parameter);

```